### ISSUE TITLE
defaultDate Option interface does not support 'multiple' or 'range' modes

### ISSUE THREAD

**Issue by MHL-Galen** [mezoistvan/ng2-flatpickr#73]
The FlatpickrOptions interface currently sets defaultDate type as ```string | Date```. For 'multiple' and 'range' type instances the default date needs to be an array of strings or dates - see [Flatpickr options](https://flatpickr.js.org/options/). 

In my local instance I have changed the defaultDate interface to accept ```string | string[] | Date | Date[]``` and can confirm that it works with date ranges but it would be good to see this implemented officially.

**Comment by mezoistvan** [comment]
The FlatpickrOptions interface is unnecessary, should be removed and replaced with the BaseOptions interface from flatpickr.

**Comment by maljukan** [comment]
Yep, all fine by using `FlatpickrOptions`, except you must declare it as `Partial<FlatpickrOptions>` as `FlatpickrOptions` are all probably generated as required: 
```typescript  
public flatpickrOptions: Partial<BaseOptions> = {
    inline: true,
    mode: 'multiple',
    defaultDate: this.multiDates
  };
```

**Comment by abdulbasit1248** [comment]
I found a fix for this. Basically, I was using two packages in my rails application. The problem is that the stimulus searches for the respective controller name. If we will not provide it gets confused as that which controller is targeting the data attributes.

```
"stimulus": "^2.0.0",
"stimulus-flatpickr": "^1.4.0",
```

```
<div class="form-group flatpicker">
      <%= f.label :days_closed, class: 'form-control-label font-weight-bold' %><br>
      <%= f.text_field :days_closed, class: 'form-control flatpickr-input active selected_date', data: {
        mode: 'multiple',
        behavior: 'flatpickr_date',
        controller: "flatpickr",
        flatpickr_date_format: "Y-m-d",
        flatpickr_alt_input: true,
        flatpickr_default_date: month_goal.days_closed,
        alt_input: true } %>
</div>

```
